examples/wasm/pull.rs

```diff
@@ -0,0 +1,24 @@
+use oci_distribution::{manifest, secrets::RegistryAuth, Client, Reference};
+use tracing::info;
+
+pub(crate) async fn pull_wasm(
+    client: &mut Client,
+    auth: &RegistryAuth,
+    reference: &Reference,
+    output: &str,
+) {
+    info!(?reference, ?output, "pulling wasm module");
+
+    let image_content = client
+        .pull(reference, auth, vec![manifest::WASM_LAYER_MEDIA_TYPE])
+        .await
+        .expect("Cannot pull Wasm module")
+        .layers
+        .into_iter()
+        .next()
+        .map(|layer| layer.data)
+        .expect("No data found");
+
+    std::fs::write(output, image_content).expect("Cannot write to file");
```

Comment:
Nit (because this is just an example): This is blocking IO in an async function. An async compatible read should be used instead

Reply:
Good point, fixed